GLAA.Web/Controllers/EligibilityController.cs

```diff
@@ -10,157 +10,225 @@ namespace GLAA.Web.Controllers
-        public ActionResult WhatDoINeed()
+        public IActionResult Back(int submittedPageId)
         {
-            return View();
+            return CheckParentValidityAndRedirectBack(submittedPageId);
         }
 
         [HttpGet]
-        public ActionResult WhatIsCovered()
+        public ActionResult Introduction()
         {
-            return View();
+            return View("Introduction");
         }
 
         [HttpGet]
-        public ActionResult OverseasBusiness()
+        public ActionResult VerificationSent()
         {
-            return View();
+            return View("VerificationSent", session.GetString(CurrentPaEmail));
         }
 
         [HttpGet]
-        public ActionResult Introduction()
+        public IActionResult ResendVerification()
         {
-            var licenceApplicationModel = licenceApplicationViewModelBuilder.New();
-
-            return View("Introduction", licenceApplicationModel);
+            accountCreationPostDataHandler.SendConfirmation(session.GetString(CurrentPaEmail), Url);
+            return RedirectToAction("VerificationSent");
         }
 
         [HttpPost]
         [ExportModelState]
-        public ActionResult Introduction(LicenceApplicationViewModel model)
+        public IActionResult SaveEmailAddress(PrincipalAuthorityEmailAddressViewModel model)
         {
-            model.NewLicenceStatus = new LicenceStatusViewModel
+            session.SetSubmittedPage(FormSection.Eligibility, 1);
+
+            if (accountCreationPostDataHandler.Exists(model.EmailAddress))
+            {
+                ViewData["doOverride"] = true;
+                ModelState.AddModelError("EmailAddress", "A user with this email address already exists in the system.");
+            }
+
+            if (!ModelState.IsValid)
+            {
+                return View(GetViewPath(FormSection.Eligibility, 1), model);
+            }
+
+            // Don't overwrite an unconfirmed user if we're currently editing that user
+            if (!model.EmailAddress.Equals(session.GetString(CurrentPaEmail),
+                StringComparison.InvariantCultureIgnoreCase))
             {
-                Id = constantService.NewApplicationStatusId
-            };
+                accountCreationPostDataHandler.DeleteIfUnconfirmed(model.EmailAddress);
+            }
 
-            var licenceId = licenceApplicationPostDataHandler.Insert(model);
+            session.SetString(CurrentPaEmail, model.EmailAddress);
 
-            session.SetCurrentLicenceId(licenceId);
+            accountCreationPostDataHandler.Update(model.EmailAddress, model);
 
-            return RedirectToAction("Part1");
+            return CheckParentValidityAndRedirect(1);
         }
 
         [HttpPost]
         [ExportModelState]
-        [Route("Eligibility/Part/1")]
-        public ActionResult Part1(SuppliesWorkersViewModel model)
+        public IActionResult SaveFullName(PrincipalAuthorityFullNameViewModel model)
```

Comment:
Much prefer these action names!

Reply:
The only disadvantage is it cocks up the nice URLs if you are sent back to the page by the server-side validation. It's probably only me that cares about that though 😉